src/stdx/bounded_array.zig

```diff
@@ -0,0 +1,68 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+/// A version of `std.BoundedArray` with TigerBeetle-idiomatic APIs.
+pub fn BoundedArray(comptime T: type, comptime capacity: usize) type {
+    const Inner = @import("std").BoundedArray(T, capacity); // smuggle the std version past tidy
+
+    return struct {
+        inner: Inner = Inner{},
+
+        const Self = @This();
+
+        pub inline fn from_slice(items: []const T) error{Overflow}!Self {
+            return .{ .inner = try Inner.fromSlice(items) };
+        }
+
+        pub inline fn count(array: *const Self) usize {
```

Comment:
Not sure whethere we use `fn count` or `fn len`, but I think it's count, based on RingBuffer, and the fact that `"count".len == "index".len`